sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/handler/ReceiverUnsettledDeliveries.java

```diff
@@ -291,14 +325,31 @@ public Mono<Void> terminateAndAwaitForDispositionsInProgressToComplete() {
     public void close() {
         isTerminated.getAndSet(true);
 
-        // Disposes of timeoutTimer's internal subscription to the global interval timer.
-        timoutTimer.dispose();
+        if (settleOnClose) {
+            // Settle unsettled deliveries to remove them from receive-link's parent ProtonJ TransportSession.
+            //
+            if (!deliveries.isEmpty()) {
+                final Runnable localSettlement = () -> {
+                    for (Delivery delivery : deliveries.values()) {
+                        delivery.disposition(new Modified());
+                        delivery.settle();
+                    }
+                };
+
+                try {
+                    dispatcher.invoke(localSettlement);
+                } catch (IOException e) {
+                    logger.warning("IO sink was closed when scheduling local settlement. Manually settling.", e);
```

Comment:
I wonder if this can be info since we handle it

Reply:
Agree, this should be _info_.